OurUmbraco/Our/SearchResultExtensions.cs

```diff
@@ -15,6 +18,25 @@ public static string GetTitle(this SearchResult result)
             return HttpContext.Current.Server.HtmlEncode(result.Fields.ContainsKey("nodeName") ? result["nodeName"] : string.Empty);
         }
 
+        public static IEnumerable<SearchResultBreadcrumbModel> GetBreadcrumbs(this SearchResult result)
+        {
+            var currentResult = new UmbracoHelper(UmbracoContext.Current).Content(result.Id) as IPublishedContent;
```

Comment:
I see a few issues with this line:

- Creating a new `UmbracoHelper` instance has a slight overhead, so initializing more than one per request should generally be avoided. In this line, an instance will be created for each search result. And the search seems to show up to 100 results, meaning a potential of 100 new instances. As our only use of `UmbracoHelper` is getting the IPublishedContent, we can do that directly via `UmbracoContext` instead - eg. by changing the line to:

    ```csharp
    var currentResult = UmbracoContext.Current.ContentCache.GetById(result.Id);
    ```

- Not all search results are based on content. For instance, forum topics are stored in the database, so when attempting to lookup a topic in the content cache, you will get `null` back - which will then result in an exception when trying to get the ancestors of `null` on the next line.

    So ideally, there should be a line line the one below between line 23 and line 24:

    ```csharp
    if (currentResult == null) return new List<SearchResultBreadcrumbModel>();
    ```

    Things don't always go according to plan, so you could also end up with inconsitencies between the content cache and the Examine index - eg. if a content item has been deleted, but still exists in the Examine index. The suggested null check above should also take care of that scenario 😉

Reply:
I have made these changes